processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryConfig.java

```diff
@@ -78,6 +79,9 @@
   @JsonProperty
   private boolean forcePushDownLimit = false;
 
+  @JsonProperty
+  private boolean applyLimitPushDownToSegment = true;
```

Comment:
main reason to have this flag to allow users to go back to older behavior in case a bug is found. TBH , that was my biggest concern to have the flag.

that said, in certain pathological case, overhead of maintaining the min-max heap might be more costly than savings from that, so it could be disabled. but is true for pushing down limits feature in general which is always on at least in the merge phase.